sys-boot/udk/udk-2018.ebuild

```diff
@@ -0,0 +1,256 @@
+	fi
+}
+
+src_configure() {
+	# Compile of Base Tools is required for further setting up the environment
+	# Base tools does not like parallel make
+	local cflags_save=${CFLAGS}
+	append-cflags -MD -fshort-wchar -fno-strict-aliasing
+	append-cflags -nostdlib -c -fPIC
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS}:" \
+		-i "${S}/BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+	CFLAGS=${cflags_save}
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	emake "${make_flags[@]}" -j1 -C BaseTools
+	. edksetup.sh
+
+	# Update UDK parameter files
+	sed -e "s:^\(ACTIVE_PLATFORM\s*=\).*$:\1 MdeModulePkg/MdeModulePkg.dsc:" \
+		-e "s:^\(TARGET\s*=\).*$:\1 RELEASE:" \
+		-e "s:^\(TARGET_ARCH\s*=\).*$:\1 ${ARCH}:" \
+		-e "s:^\(TOOL_CHAIN_TAG\s*=\).*$:\1 ${TOOLCHAIN_TAG}:" \
+		-e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+		-i "${S}/Conf/target.txt" || die "Failed to configure target file"
+	sed -e "s:\(_CC_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASM_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_PP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_VFRPP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLCC_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLPP_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_DLINK_PATH\s*=\).*$:\1 $(tc-getCC):" \
+		-e "s:\(_ASLDLINK_PATH\s*=\).*$:\1 $(tc-getLD):" \
+		-e "s:\(_OBJCOPY_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_RC_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+		-e "s:\(_SLINK_PATH\s*=\).*$:\1 $(tc-getAR):" \
+		-i "${S}/Conf/tools_def.txt" \
+		|| die "Failed to prepare tools definition file"
+}
+
+src_compile() {
+	local build_target
+	if use examples; then
+		build_target=all
+	else
+		build_target=libraries
+	fi
+
+	build ${build_target} || die "Failed to compile environment"
+
+# TODO * /var/tmp/portage/sys-apps/sandbox-2.10-r1/work/sandbox-2.10/libsandbox/trace.c:_do_ptrace():74: failure (Operation not permitted):
+# TODO * ISE:_do_ptrace: ptrace(PTRACE_TRACEME, ..., 0x0000000000000000, 0x0000000000000000): Operation not permitted
+}
+
+src_install() {
+	local f
+	local build_dir="${S}/Build/MdeModule/RELEASE_${TOOLCHAIN_TAG}/${ARCH}"
+
+	for f in "${build_dir}"/*/Library/*/*/OUTPUT/*.lib; do
+		newlib.a "${f}" lib$(basename "${f}" .lib).a
+	done
+	
+	insinto "usr/share/${P}"
+	doins "${S}/BaseTools/Scripts/GccBase.lds"
+
+	local include_dest="/usr/include/${PN}"
+	for f in "" /Guid /IndustryStandard /Library /Pi /Ppi /Protocol /Uefi; do
+		insinto "${include_dest}${f}"
+		doins "${S}/MdePkg/Include${f}"/*.h
+	done
+	insinto "${include_dest}"
+	doins "${S}/MdePkg/Include/${ARCH}"/*.h
+	local hfile
+	find "${S}" -name 'BaseTools' -prune -o -name 'MdePkg' -prune -o \
+		-name 'CryptoPkg' -prune -o -type d -name Include \
+		-exec find {} -maxdepth 0 \; \
+		| while read hfile; do
+		doins -r "${hfile}"/*
```

Comment:
`/.` instead of `/*`, to make it install everything at once instead of one by one.